node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -283,37 +305,51 @@ impl Initialized {
 		self.participation.process_active_leaves_update(ctx, &update).await?;
 
 		if let Some(new_leaf) = update.activated {
-			match self
-				.rolling_session_window
-				.cache_session_info_for_head(ctx.sender(), new_leaf.hash)
-				.await
-			{
-				Err(e) => {
-					gum::warn!(
+			let session_idx =
+				self.runtime_info.get_session_index_for_child(ctx.sender(), new_leaf.hash).await;
+
+			match session_idx {
+				Ok(session_idx) if session_idx > self.last_consecutive_cached_session => {
+					// There is a new session. Perform a dummy fetch to cache it.
+					let mut gap_in_cache = false;
+					for idx in self.last_consecutive_cached_session + 1..=session_idx {
```

Comment:
If we fail to fetch a session info on startup, `last_consecutive_cached_session` will be initialized as 0 and this loop will be a very long one, right?

Reply:
That's a very good catch. Indeed it will be one pretty long loop.